Feature/2/add orders

RyszardKluz opened the pull request.

RyszardKluz committed:
Added modules for clients, products and orders, created controllers and routes for orders. Started working on client-side logic for orders.

RyszardKluz committed:
created checbox component for list of clients, created few components for orders, and added basic logic for modal and orders page.

RyszardKluz committed:
Refactored clients feature with Service-Repository pattern: improved structure, error handling, and consistency,

RyszardKluz committed:
"Added checkbox"

RyszardKluz committed:
Merge branch 'feature/2/add-orders' of https://github.com/RyszardKluz/order-management into feature/2/add-orders

pbujanowski commented on client/src/components/Checkbox.jsx:
```diff
@@ -0,0 +1,11 @@
+const Checkbox = ({ handleClick }) => {
+  return (
+    <>
+      <Form.Check onClick={handleClick} type="radio" aria-label="radio 1" />
```
If we have shared component, like this, I suggest to pass additional value for `aria-label` (right now, all checkboxes will have that attribute set to `radio 1`) or completely remove that attribute, at least for now. It's not necessary, it just improve accessibility for users.

pbujanowski commented on client/src/pages/orders/components/NewOrderModal.jsx:
```diff
@@ -0,0 +1,34 @@
+const newOrderModal = ({ isVisible, onClose }) => {
+  return (
+    <>
+      {' '}
```
Is it leftover or it's necessary?

pbujanowski reviewed (commented):
Overall looks fine. I suggest to merge that PR before #3.

RyszardKluz committed:
Merge remote-tracking branch 'origin/master' into feature/2/add-orders

RyszardKluz committed:
Added order details and product counting logic

- Implemented order details management and product counting features.
- Created helper functions for toggling the active key in the accordion and calculating the total amount.
- TODO: Fix issue with multiple instances of the same object in OrderDetailsList.
- Add error handling to new features.

pbujanowski commented on client/src/components/Lists/ResourceList.jsx:
```diff
@@ -28,16 +49,17 @@ const ResourceList = ({
     <ResourceListTable
       columnHeadings={columnHeadings}
       handleRowClick={handleRowClick}
+      handleCheckboxClick={handleCheckboxClick}
```
I think it could be better to change properties names from `handle` to `on`, but in component (here will be `ResourceListTable`). For functions, which are for handling those events, I will leave `handle` prefix.

RyszardKluz replied:
Oh, yeah, I did stick to that naming pattern before, didn't notice this before. Tnx

RyszardKluz committed:
feat(order):
- finish client-side logic for creating new orders
- add server-side validation for order and products

TODO:
- client-side error handling
- fix server-side error sending logic

RyszardKluz committed:
feat(order): improve error handling

- Fix server-side error handling logic
- Improve client-side error handling
-Minor backend code refactoring

TODO:
- Add client-side logic for fetching orders
- Review backend implementation of GET /orders

RyszardKluz committed:
feat(order) : minor improvement of error handling

pbujanowski commented on client/src/helpers/validateInputData.js:
```diff
@@ -1,4 +1,11 @@
 const validateInputData = (inputData, inputFields) => {
+
+
+  for (const [key] of Object.entries(inputData)) {
```
I suggest using `Prettier` for format code.

RyszardKluz replied:
Done 🥇

pbujanowski commented on server/src/services/OrdersService.js:
```diff
@@ -3,20 +3,28 @@ import AppError from '../errors/AppError.js';
+    //   typeof body.productName !== 'string'
+    // ) {
+    //   throw new AppError('Invalid product price or name format', 400);
+    // }
```
Probably it needs to be removed.

RyszardKluz replied:
Yup, I was trying something and forgot to remove that, thanks 👍

RyszardKluz committed:
feat (order) :
-format code
-fix problem with props name

RyszardKluz committed:
feat(order):
-add logic for rendering order list
-correct totalPrice calculation
-improve order dispatch to frontend
